### PR TITLE
Lps 107626

### PR DESCRIPTION
**Pull request by shuyangzhou**

### CODE REVIEW

**Comment by brianchandotcom** [comment]
Merged. Thank you.
View total diff: https://github.com/brianchandotcom/liferay-portal/compare/9808a4a2a0...47ff636ccf

**Comment by brianchandotcom** [comment]
@ivicac I backported and published this for you all the way to 7.1.x

@marcoleo fyi

**Comment by brianchandotcom** [comment]
@nhpatt can you send the batch generation now that this is in? Thx.

**Comment by nhpatt** [comment]
I have to rewrite it but yeah, I'll work on it now.

**Comment by ivicac** [comment]
thx @brianchandotcom

File: modules/apps/batch-engine/batch-engine-test/src/testIntegration/java/com/liferay/batch/engine/internal/test/BaseBatchEngineTaskExecutorTest.java [hunk 1]
```diff
@@ -169,27 +168,16 @@ public void createItem(
...
+
+			Image image = blogPosting.getImage();
 
 			blogsEntryService.addEntry(
 				blogPosting.getHeadline(), blogPosting.getAlternativeHeadline(),
 				blogPosting.getFriendlyUrlPath(), blogPosting.getDescription(),
 				blogPosting.getArticleBody(), localDateTime.getMonthValue() - 1,
 				localDateTime.getDayOfMonth(), localDateTime.getYear(),
 				localDateTime.getHour(), localDateTime.getMinute(), true, true,
-				new String[0],
```
Comment (ivicac) [inline modules/apps/batch-engine/batch-engine-test/src/testIntegration/java/com/liferay/batch/engine/internal/test/BaseBatchEngineTaskExecutorTest.java]: hi @shuyangzhou 
this is taken from the original source. Should I ping code owners to update code according to your changes?

Best Regards

File: modules/apps/batch-engine/batch-engine-test/src/testIntegration/java/com/liferay/batch/engine/internal/test/BaseBatchEngineTaskExecutorTest.java [hunk 2]
```diff
@@ -169,27 +168,16 @@ public void createItem(
...
-					Image::getCaption
-				).orElse(
-					null
-				),
-				_getImageSelector(
-					imageOptional.map(
-						Image::getImageId
-					).orElse(
-						null
-					)),
```
Comment (ivicac) [inline modules/apps/batch-engine/batch-engine-test/src/testIntegration/java/com/liferay/batch/engine/internal/test/BaseBatchEngineTaskExecutorTest.java]: ivicac now
hi @shuyangzhou
this is taken from the original source. Should I ping code owners to update code according to your changes?

Best Regards

File: modules/apps/batch-engine/batch-engine-test/src/testIntegration/java/com/liferay/batch/engine/internal/test/BaseBatchEngineTaskExecutorTest.java [hunk 3]
```diff
@@ -262,18 +251,7 @@ public void updateItem(
...
```
Comment (ivicac) [inline modules/apps/batch-engine/batch-engine-test/src/testIntegration/java/com/liferay/batch/engine/internal/test/BaseBatchEngineTaskExecutorTest.java]: the same as above

**Comment by shuyangzhou** [comment]
@ivicac Stream and Optional usages should be restricted to only necessary places. They are bad for performance.

**Comment by shuyangzhou** [comment]
@ivicac yeah, go ahead to update the original locations or let the owner know.

**Comment by ivicac** [comment]
@nhpatt can you please update BlogPostingResourceImpl according to changes Shuyang made in my test for Batch(I used your original code from BlogPostingResourceImpl for my Batch test and Shuyang made some changes).

Thank you

**Pull request closed without merging**